### DISCUSSION TITLE
Discussed Time Series Features

### DISCUSSION THREAD

**Discussion by Gerhardsa0** [Safe-DS/.github#6] [Library Discussion]
When working with time series, you normally always want a time column, which can be the dates or just normal indices called time dummies.
Right now, a user always needs to give specific a time column name, but should a user also be possible to not give a time name, and a time dummie column will generate automatically.

**Comment by Gerhardsa0** [comment]
Should a time series have the feature to be shuffled? Because the data has an order, which is also important for a nn model to learn, and I think also statistical models don't want/need the data to be shuffled. The same for sorting rows, because they should always be sorted by the date.

**Comment by Gerhardsa0** [comment]
Should the time column be transformable?

**Reply by lars-reimann** [reply]
From the perspective of a table it's a normal column. So if the transformation can handle temporal data, there's no reason to exclude it.

**Comment by lars-reimann** [comment]
Based on those questions, you should also consider whether a `TimeSeries` should really inherit from `Table`.

**Reply by Gerhardsa0** [reply]
Yes, I rly dont know a the Moment
TimeSeries shares features with TaggedTable and I come to some difficulties.

Right now, when the time_column is a feature in the Tagged_table it will automatically be removed from the features and marked as a time_column in the from_tagged_table()-function.
Is this behavior wanted?

**Reply by lars-reimann** [reply]
As long as you don't mutate the original `TaggedTable`, it's OK.
